src/site/content/en/payments/android-payment-apps-delegation/index.md

```diff
@@ -1,36 +1,267 @@
+  UI when asking for the user's address or choice of shipping options.
+
+### `shippingOptions` {: #shipping-options }
+
+`shippingOptions` is the parcelable array of merchant specified shipping
+options. This parameter will only exist when `paymentOptions.requestShipping ==
+true`.
+
+```kotlin
+val shippingOptions: List<ShippingOption>? =
+    extras.getParcelableArray("shippingOptions")?.mapNotNull {
+        p -> from(p as Bundle)
+    }
+```
+
+Each shipping option is a `Bundle` with the following keys.
+
+* `id` - The shipping option identifier.
+* `label` - The shipping option label shown to the user.
+* `amount` - The shipping cost bundle containing `currency` and `value` keys with
+  string values.
+    * `currency` shows the currency of the shipping cost, as an
+      [ISO4217](https://www.iso.org/iso-4217-currency-codes.html) well-formed
+      3-letter alphabet code
+    * `value` shows the value of the shipping cost, as a [valid decimal monetary
+      value](https://w3c.github.io/payment-request/#dfn-valid-decimal-monetary-value)
+* `selected` - Whether or not the shipping option should be selected when the
+  payment app displays the shipping options.
+
+All keys other than the `selected` have string values. `selected` has a boolean
+value.
+
+```kotlin
+val id: String = bundle.getString("id")
+val label: String = bundle.getString("label")
+val amount: Bundle = bundle.getBundle("amount")
+val selected: Boolean = bundle.getBoolean("selected", false)
+```
+
+## Provide required information in a payment response
+
+Your app should include the required additional information in its response to
+the `PAY`  activity.
+
+To do so the following parameters must be specified as Intent extras:
+
+* `payerName` - The payer's full name. This should be a non-empty string when
+  `paymentOptions.requestPayerName` is true.
+* `payerPhone` - The payer's phone number. This should be a non-empty string when
+  `paymentOptions.requestPayerPhone` is true.
+* `payerEmail` - The payer's email address. This should be a non-empty string
+  when `paymentOptions.requestPayerEmail` is true.
+* `shippingAddress` - The user-provided shipping address. This should be a
+  non-empty bundle when `paymentOptions.requestShipping` is true. The bundle
+  should have the following keys which represent different parts in a [physical
+  address](https://www.w3.org/TR/payment-request/#physical-addresses).
+    * `city`
+    * `countryCode`
+    * `dependentLocality`
+    * `organization`
+    * `phone`
+    * `postalCode`
+    * `recipient`
+    * `region`
+    * `sortingCode`
+    * `addressLine`
+  All keys other than the `addressLine` have string values. The `addressLine`
+  is an array of strings.
+* `shippingOptionId` - The identifier of the user-selected shipping option. This
+  should be a non-empty string when `paymentOptions.requestShipping` is true.
+
+###  Payment response validation
```

Comment:
This can be "Validate payment response" to stay consistent with the other headers